src/main/java/com/github/susom/database/DatabaseProviderVertx.java

```diff
@@ -668,6 +671,9 @@ public void rollbackAndClose() {
 
     if (connection != null) {
       try {
+        if (options.flavor().autoCommitOnly()) {
+          throw new UnsupportedOperationException("rollback is not supported");
+        }
```

Comment:
Why silently ignore commit, but raise an exception on rollback?

Reply:
Database-agnostic request-handling code tries to call commitAndClose in the normal case or rollbackAndClose in the exceptional case.  I'm open to suggestions.

Reply:
The APIs are all designed with the assumption of transaction control, so if we run with a database that doesn't support it, we have to understand our semantics are broken. I  would be inclined to make this rollback silently ignore like lines 648-650, because it is more compatible (avoid failures in the default rollback on errors), and at this point in the code any damage is already done. If we want to raise an error, we should probably have a separate API for auto commit "transact()" code blocks, and raise exceptions at the beginning before we allow any statements to be executed.

Reply:
Another reason silently ignoring might be good: I can write my code as if transactions work, and maybe in a few years when the database/driver add support, we can flip the flag in `Flavor` and it will start working without client code changes.

Reply:
Agreed.